File: apps/api/FinanceControl.Shared/Dtos/Response/WishlistItemResponseDto.cs
using FinanceControl.Shared.Enums;

namespace FinanceControl.Shared.Dtos.Response
{
    public class WishlistItemResponseDto
    {
        public int Id { get; init; }
        public string Name { get; init; } = string.Empty;
        public string? Description { get; init; }
        public int? TargetPrice { get; init; }
        public EnumWishlistPriority Priority { get; init; }
        public EnumWishlistStatus Status { get; init; }
        public string? Url { get; init; }
        public string? ImageUrl { get; init; }
        public int? LatestPrice { get; init; }
        public DateTime CreatedAt { get; init; }
        public DateTime? UpdatedAt { get; init; }
    }
}

File: apps/mobile/lib/features/wishlist/data/dtos/wishlist_item_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'wishlist_item_response_dto.g.dart';

@JsonSerializable(createToJson: false)
class WishlistItemResponseDto {
  const WishlistItemResponseDto({
    required this.id,
    required this.name,
    this.description,
    this.targetPrice,
    required this.priority,
    required this.status,
    this.url,
    this.imageUrl,
    this.latestPrice,
    required this.createdAt,
    this.updatedAt,
  });

  final int id;
  final String name;
  final String? description;
  final int? targetPrice;
  final String priority;
  final String status;
  final String? url;
  final String? imageUrl;
  final int? latestPrice;
  final DateTime createdAt;
  final DateTime? updatedAt;

  factory WishlistItemResponseDto.fromJson(Map<String, dynamic> json) =>
      _$WishlistItemResponseDtoFromJson(json);
}
